### PR TITLE
move assertions out of system

### PR DESCRIPTION
**Pull request by ringabout**
ref https://github.com/nim-lang/RFCs/issues/437

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -1830,8 +1830,11 @@ when not defined(nimscript):
 when defined(nimV2):
   include system/arc
 
-import system/assertions
-export assertions
+when not defined(nimPreviewSlimSystem):
+  {.deprecated: """assertions is about to move out of system; use `-d:nimPreviewSlimSystem` and
+                import `std/assertions`.""".}
+  import std/assertions
+  export assertions
```
Comment (ringabout) [inline lib/system.nim:1837]: review here

File: lib/std/assertions.nim [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2022 Nim contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## This module implements assertion handling.
```
Comment (ringabout) [inline lib/std/assertions.nim:10]: review here

**Comment by ringabout** [comment]
Except the two files I mentioned above, others changes are about import std/assertions(syncio) under `-d:nimPreviewSlimSystem` condition.

I use command `koch temp --lib:lib -d:nimPreviewSlimSystem` for now. Following PRs will use `koch boot` to check the compiler.

File: lib/std/assertions.nim (hunk 2 above)
Comment (Araq) [reply lib/std/assertions.nim:10]: It's fine.

File: lib/system.nim (hunk 1 above)
Comment (Araq) [reply lib/system.nim:1837]: This is a bad idea, users have no control over system.nim. Instead remove the import/export of assertions and patch the tests. Important packages can be compiled via `--useVersion:1.6` or similar.

Comment (ringabout) [reply lib/system.nim:1837]: Ok, here is the first step: https://github.com/nim-lang/Nim/pull/19601 (support useVersion:1.6)

Comment (ringabout) [reply lib/system.nim:1837]: Is something like this okay?

https://github.com/nim-lang/Nim/pull/19602

**Pull request merged**